source/Calamari/Integration/Processes/Semaphores/SystemSemaphoreManager.cs

```diff
@@ -22,9 +22,15 @@ public SystemSemaphoreManager(ILog log, TimeSpan initialWaitBeforeShowingLogMess
             this.initialWaitBeforeShowingLogMessage = (int)initialWaitBeforeShowingLogMessage.TotalMilliseconds;
         }
 
-#if NET40
         public IDisposable Acquire(string name, string waitMessage)
         {
+            return CalamariEnvironment.IsRunningOnWindows
+                ? AcquireSemaphore(name, waitMessage)
+                : AcquireMutex(name, waitMessage);
```

Comment:
Yeh that would be good (this sparked my #feature-discussion thread). I think the initial approach was not to change functionality in the existing code, but this sounds like a good change.